lua/xcodebuild/helpers.lua

```diff
@@ -95,6 +95,24 @@ function M.get_major_os_version()
   return settings.os and tonumber(vim.split(settings.os, ".", { plain = true })[1]) or nil
 end
 
+---Wraps any nvim_buf_set_option() call to ensure backward and forward compatibility.
+---The function is required because nvim_buf_set_option was deprecated in nvim-0.10.
+---@param bufOrWinID number
+---@param name string
+---@param value any
+---@param win boolean
+function M.nvim_buf_or_win_set_option_fwd_comp(bufOrWinID, name, value, win)
```

Comment:
To make it more clear when reading code, I would create two functions:
`function set_buf_option(bufnr, name, value)`
and
`function set_win_option(winnr, name, value)`